libraries/features/profiler/bsg_manycore_profiler.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef __BSG_MANYCORE_PROFILER_HPP
+#define __BSG_MANYCORE_PROFILER_HPP
+#include <bsg_manycore.h>
+#include <string>
+
+// Since the definition of hb_mc_profiler_t is implementation
+// dependent, we use void *
+typedef void* hb_mc_profiler_t;
```

Comment:
This is interesting. Where are the actual contents defined in a platform usually?

Reply:
Are you asking, what does the void * point to? 

In a simulation platform, it points to a std::vector: 

https://github.com/bespoke-silicon-group/bsg_replicant/blob/18d4ccdcc88dbc6e482bef2bf318bcf87f799dbc/libraries/features/profiler/simulation/bsg_manycore_profiler.cpp#L58

I used a void * so that another platform that wants to implement a profiler could design its own implementation.

Reply:
I makes sense. That, host code wouldn't have to worry about managing profiler but can just get info by calling get_icount etc. at will.

Reply:
Great! The details of the contents of void* are abstracted by api functions (like get_icount and get_cycles)?

Reply:
Yep - https://github.com/bespoke-silicon-group/bsg_replicant/blob/18d4ccdcc88dbc6e482bef2bf318bcf87f799dbc/examples/cuda/test_vec_add.c#L171

(Something didn't get pushed up, those should be enum values not raw integers)